include/alpaka/mem/buf/BufCpu.hpp

```diff
@@ -233,7 +225,39 @@ namespace alpaka
             {
                 ALPAKA_DEBUG_MINIMAL_LOG_SCOPE;
 
-                return BufCpu<TElem, TDim, TIdx>(dev, extent);
+                // alpaka::AllocCpuAligned is stateless
+                using Allocator
+                    = AllocCpuAligned<std::integral_constant<std::size_t, core::vectorization::defaultAlignment>>;
```

Comment:
Should we assert this? Something like:
```c++
struct Test : Allocator { int i; };
static_assert(sizeof(Test) == sizeof(int));
```

Reply:
I see, but I don't think it would be useful, because a similar `assert()` would still pass for an `Allocator` that holds _e.g._ a pointer to an internal buffer, etc.

Reply:
Ah, no, sorry, I misread the example.

Reply:
What about
```c++
static_assert(std::is_empty_v<Allocator>, "AllocCpuAligned is expected to be stateless");
```
?

Reply:
Perfect! I did not know about that trait :) Thx!